packages/pattern-picker/src/use-link-in-bio-designs.ts

```diff
@@ -0,0 +1,48 @@
+import { useStarterDesignsQuery } from '@automattic/data-stores';
+import { useLocale } from '@automattic/i18n-utils';
+import { useQuery } from '@tanstack/react-query';
+import type { Design } from '@automattic/design-picker';
+
+/**
+ * This query isn't really a query, it's just meant to use Calypso's react-query async storage to store the order for the user forever
+ *
+ * @param length the length of the array to order
+ * @returns an array of numbers in a random order
+ */
+function useOrder( length: number ) {
+	return useQuery(
+		[ 'ptk/order', length ],
+		async () => {
+			const indices = Array.from( { length }, ( _, i ) => i );
+			const firstFour = indices.slice( 0, 4 ).sort( () => Math.random() - 0.5 );
+			const theRest = indices.slice( 4 );
+			return [ ...firstFour, ...theRest ];
+		},
+		{
+			staleTime: Infinity,
+			enabled: length > 0,
+		}
+	);
+}
+
+export const useLinkInBioDesigns = (): Design[] => {
+	const queryResult = useStarterDesignsQuery( {
+		_locale: useLocale(),
```

Comment:
Optional: you can pass `select` and `initialData` here to filter instead. 
```suggestion
		_locale: useLocale(),
		initialData: [],
		select ( data ) {
			return data.designs.filter(
				( design ) =>
					design.is_virtual &&
					design.categories.some( ( category ) => category.slug === 'link-in-bio' )
		}
```

Reply:
I thought about that too, but `useStarterDesignsQuery` has a custom `select` property which complicates things a little bit because of the `StarterDesigns ` type:

https://github.com/Automattic/wp-calypso/blob/defe650653380df86a22df68d695b652041e5178/packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts#L21

https://github.com/Automattic/wp-calypso/blob/defe650653380df86a22df68d695b652041e5178/packages/data-stores/src/starter-designs-queries/use-starter-designs-query.ts#L47-L56

So I finally decided to filter the given result.